src/main/java/graphql/execution/ValuesResolverOneOfValidation.java

```diff
@@ -75,23 +75,36 @@ static void validateOneOfInputTypes(GraphQLType type, Object inputValue, Value<?
     }
 
     private static void validateOneOfInputTypesInternal(GraphQLInputObjectType oneOfInputType, Value<?> argumentValue, Map<String, Object> objectMap, Locale locale) {
-        int mapSize;
-
+        final String fieldName;
         if (argumentValue instanceof ObjectValue) {
-            mapSize = ((ObjectValue) argumentValue).getObjectFields().size();
+            List<ObjectField> objectFields = ((ObjectValue) argumentValue).getObjectFields();
+            if (objectFields.size() != 1) {
+                throwNotOneFieldError(oneOfInputType, locale);
+            }
+
+            fieldName = objectFields.iterator().next().getName();
         } else {
-            mapSize = objectMap.size();
-        }
-        if (mapSize != 1) {
-            String msg = I18n.i18n(I18n.BundleType.Execution, locale)
-                    .msg("Execution.handleOneOfNotOneFieldError", oneOfInputType.getName());
-            throw new OneOfTooManyKeysException(msg);
+            if (objectMap.size() != 1) {
+                throwNotOneFieldError(oneOfInputType, locale);
+            }
+
+            fieldName = objectMap.keySet().iterator().next();
         }
-        String fieldName = objectMap.keySet().iterator().next();
```

Comment:
The bug was in this line. We were checking the size of `argumentValue.getObjectFields()` but trying to get the first element of `objectMap`, which would be empty if no variables were passed.

Reply:
if (argumentValue instanceof ObjectValue)

we got mapSize from argumentValue. But objectMap can still be empty and we didn't check objectMap size. 

That's the problem , right?

Reply:
Yep, that's right